src/firmware/Firmware-IGVC-2018/Firmware-IGVC-2018.ino

```diff
@@ -0,0 +1,146 @@
+/* Created May 5th by Chris Heathe
+ *  
+ * Basic arduino motor driver that subsribes to the /cmd_vel topic 
+ * and prints corresponding servo values to the left and write motors 
+ * of Elsa.
+ * 
+ * PID loop has now been integrated, Kp, Ki, Kd values will need to be calibrated in physical testing
+ * The library that was used can be found at:
+ * 
+ * https://playground.arduino.cc/Code/PIDLibrary
+ * 
+ * !!NOTES ON USING THE ros.h ARDUINO LIBRARY!!
+ * 
+ * Install the 'ros-kinetic-rosserial-server' package
+ * 
+ * Once installed it is necessary to run the following command in terminal
+ * to establish a serial connection and ROS:
+ * 
+ * rosrun rosserial_python serial_node.py /dev/ttyACM0
+ * 
+ */
+
+
+#include <ros.h>
+#include <geometry_msgs/Twist.h>
+#include <nav_msgs/Odometry.h>
+#include<Servo.h>
+#include<PID_v1.h>
+
+
+Servo leftMotor;
+Servo rightMotor;
+const int leftMotorPin = 10;
+const int rightMotorPin = 11;
+
+
+// Will be used to initialized the motors so that fluctuations in printing will not create jerky movements
+int minPWM = 1000;
+int maxPWM = 2000;
+
+
+// Maximum and minimum values we can write to the motor
+int maxMotorWrite = 180;
+int minMotorWrite = 0;
+
+// The numbers we want to print to the motor ranging from 0 to 180 with 90 as the centerpoint (not moving)
+double setLinear = 90;
+double setAngular = 90;
+
+// The readings we receive from the encoders ranging from 0 to 180 with 90 as the centerpoint (not moving)
+double inputLinear = 90;
+double inputAngular = 90;
+
+// The value computed by the PID function from the set and input velocities ranging from 0 to 180 with 90 as the centerpoint (not moving)
+double outputLinear = 90;
+double outputAngular = 90;
+
+// The numbers we will print to the motor based on converting PID outputs to left and right motor values
+int setLeftMotor = 90;
+int setRightMotor = 90;
+
+//Tuning parameters for the PID controllers:
+int linearKp = 2;
+int linearKi = 5;
+int linearKd = 1;
+int angularKp = 2;
+int angularKi = 5;
+int angularKd = 1;
```

Comment:
Since we're using rosserial, would it be possible to set these via rosparams? This would make it a lot easier to test, as we could quickly change values in the launch file instead of having to re-flash the arduino. See http://wiki.ros.org/rosserial/Overview/Parameters